### PR TITLE
Recording format

### PR DESCRIPTION
**Pull request by papr**
~~**This is work in progress only. Please do not merge yet!**~~ Ready to merge.

### CODE REVIEW

File: developer-docs/recording-format.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+title = "recording format"
+section_weight = 4
+page_weight = 4
++++
+
+## Recording Format
+
+### Required Files
+
+> Example `info.csv` fil
```
Comment (willpatera) [inline developer-docs/recording-format.md]: typo; `fil` --> `file`

File: developer-docs/recording-format.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+Start Time (Synced),701730.897108953
+Duration Time,00:00:13
+World Camera Frames,402
+World Camera Resolution,1280x720
+Capture Software Version,1.7.159
+Data Format Version,1.8
+System Info,"User: name, Platform: Linux, ..."
+```
+
+Each recording requires two things:
```
Comment (willpatera) [inline developer-docs/recording-format.md]: wording; `things` --> `files` or `things` --> `assets`

File: developer-docs/recording-format.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+1. An `info.csv` file that includes two columns -- `key` and `value`.
+2. At least one video file and its corresponding timestamps. See the [*Video Files*](#video-files) section below for details.
```
Comment (willpatera) [inline developer-docs/recording-format.md]: consider: `and its corresponding timestamp file`

File: developer-docs/recording-format.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+Each row of the `info.csv` file corresponds to a key-value-pair. At least two pairs are required:
```
Comment (willpatera) [inline developer-docs/recording-format.md]: Cosnider`

A minimum requirement of two key, value pairs are required. Example:

File: developer-docs/recording-format.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+1. `Recording Name`
```
Comment (willpatera) [inline developer-docs/recording-format.md]: Consider including dummy value data.

File: developer-docs/recording-format.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+2. `Data Format Version`
+
+### Data Files
+
+#### Timestamp Files
+Each data file has a corresponding timestamps file. They need to follow this *strict* naming:
```
Comment (willpatera) [inline developer-docs/recording-format.md]: Consider:

Timestamp files must follow this *strict* naming convention:

File: developer-docs/recording-format.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+Given that a media file is named `<name>.<ext>` then its timestamps file has to be named `<name>_timestamps.npy`.
+
+Timestamp files are saved in the [NPY binary format](https://docs.scipy.org/doc/numpy/neps/npy-format.html). You can use [`numpy.load()`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.load.html#numpy.load) to access the timestamps in Python.
+
+A datum and its timestamp have the same index within their respective files, i.e. the `i`th timestamp in `world_timestamps.npy` belongs to the `i`th video frame in `world.mp4`.
```
Comment (willpatera) [inline developer-docs/recording-format.md]: this is clear, nice :+1:

File: developer-docs/recording-format.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+- `.h264`
+- `.mjpeg`
+
+Allowed video file names are:
+- `world`: Scene video
+- `eye0`: Right eye video
+- `eye1`: Left eye video
+
+### Audio File
+Audio is only recognized by the playback plugin if the file is named `audio.mp4`.
```
Comment (willpatera) [inline developer-docs/recording-format.md]: Consider:

An audio file is only recognized in Pupil Player's playback plugin if the file is named `audio.mp4`.

**Comment by papr** [comment]
@willpatera No, thank you, ready to merge.

**Pull request merged**